composer.json

```diff
@@ -15,8 +15,8 @@
     ],
     "require": {
         "php": "^7.2 || ^8.0",
-        "doctrine/common": "^2.11",
-        "doctrine/persistence": "^1.3.3"
+        "doctrine/common": "^2.13|^3.0",
+        "doctrine/persistence": "^1.3.3|^2.0"
```

Comment:
It looks like none of the builds is installing versions fitting to the version constraints `^3.0` and `^2.0`. How can we be sure that the newer versions will work too?

Reply:
How can they be newer? 3 and 2 are the latest versions…

Reply:
Both doctrine/common 3 and doctrine/persistence 2 will not install on dev, because of [doctrine/mongodb-odm](https://github.com/doctrine/mongodb-odm) 1.3.x (latest version is 2.1.x) not having support for doctrine/common 3.

A few options:
1. Add support for doctrine/mongodb-odm 2.x & drop support for doctrine/mongodb-odm 1.3.x in this pull request. They don't go together nicely (tried to in a previous version of this pull request)
2. After this pull request is merged upstream, go for the first option but on the dev branch instead.
2. Replace doctrine/common in doctrine/mongodb-odm 1.3.x with the new doctrine packages like done in 2.x](https://github.com/doctrine/mongodb-odm/commit/21501c42aa50578f2cbc494a55ec4cb2995f20c1) [could something like this?](https://github.com/doctrine/data-fixtures/commit/b528aa964b5965d958f0f533df2ed611520dad5c)

Reply:
Option 3 sounds best, can you manage that?

Reply:
Worth a try ;)

Quite the task tho. A little help would be nice. Several classes (1) are no longer available due to the removal of doctrine/common. They will have to be replaced in a way, for which I've found docs https://www.doctrine-project.org/2018/07/12/common-2-9-and-dbal-2-8-and-orm-2-6-2.html

For example, to get proxying to work, we'll have to backport changes like https://github.com/doctrine/mongodb-odm/pull/1875 This results in BC breaks, which I don't think is acceptable on this stable branch.

(1) Classes no longer available;
```
Doctrine\Common\Proxy\AbstractProxyFactory
Doctrine\Common\Proxy\ProxyDefinition
Doctrine\Common\Proxy\ProxyGenerator
Doctrine\Common\Util\ClassUtils
Doctrine\Common\Util\Debug
Doctrine\Common\Proxy\Proxy
```

Will give supporting both doctrine/mongodb-odm 1.x & 2.x another try too at the end of the week.

Reply:
The classes above are still available in common v3. Why not upgrade to common 3 and replacement packages in v1 of `doctrine/mongodb-odm`?

Reply:
Also, I think there is another solution: allow `doctrine/mongodb-odm` v2 as a dev dependency of this package… I'm not sure why it isn't the case yet, any idea @alcaeus @malarzm ?

Reply:
@fruitwasp after taking a deeper look to this, I think option 1 might be best in fact… Apparently `doctrine/mongodb-odm` is just a dev dependency, which means people can already install v2 alongside doctrine fixtures

Reply:
With data-fixtures as a dependency in a project, its require-dev parts won't be affected by a v2 constraint. Suggestion 2 would be possible.

Reply:
@SenseException now that #351 is merged, we have the prefer lowest build that uses 1 and 2, while all other builds use 2 and 3.